src/LanguageServer/Impl/Implementation/Server.Documents.cs

```diff
@@ -78,5 +82,10 @@ public sealed partial class Server {
             _log?.Log(TraceEventType.Error, $"Unable to find document {uri}");
             return null;
         }
+
+        private void DocNewAstEvent(object sender, EventArgs _) {
+            var document = (IDocument)sender;
+            _indexManager.ReIndexFileAsync(document.Uri.AbsolutePath, document);
```

Comment:
Since you are not awaiting here, add DoNotWait(). Same in tests btw.

Reply:
Speaking of throttling, see DiagnosticsService, it throttles error publishing to the client.